util/dockerutil/client.go

```diff
@@ -46,17 +46,24 @@ func (c *Client) LoadImage(ctx context.Context, name string, status progress.Wri
 	w = &waitingWriter{
 		PipeWriter: pw,
 		f: func() {
-			resp, err := dapi.ImageLoad(ctx, pr, false)
-			defer close(done)
-			if err != nil {
+			discard := func(err error) {
 				pr.CloseWithError(err)
 				w.mu.Lock()
 				w.err = err
 				w.mu.Unlock()
+			}
+
+			resp, err := dapi.ImageLoad(ctx, pr, false)
+			defer close(done)
+			if err != nil {
+				discard(err)
 				return
 			}
+
 			prog := progress.WithPrefix(status, "", false)
-			progress.FromReader(prog, "importing to docker", resp.Body)
+			if err := fromReader(prog, "importing to docker", resp.Body); err != nil {
+				discard(err)
```

Comment:
With the repro above, it outputs the error in the progress output as expected and `w.err` is full filled in `Close()` and returned: https://github.com/docker/buildx/blob/b84e0e11b4384d3f51e63fe71c4be7a9aad946df/util/dockerutil/client.go#L116 but build still succeeds so it seems BuildKit discard errors on close for `ExportEntry.Output`.

Reply:
Opened https://github.com/moby/buildkit/pull/4117